src/info/infra/repositories/DataverseInfoRepository.ts

```diff
@@ -0,0 +1,27 @@
+import { IDataverseInfoRepository } from '../../domain/repositories/IDataverseInfoRepository';
+import axios, { AxiosResponse } from 'axios';
+import { ReadError } from '../../../core/domain/repositories/ReadError';
+import { DataverseVersion } from '../../domain/models/DataverseVersion';
+
+export class DataverseInfoRepository implements IDataverseInfoRepository {
+  private apiUrl: string;
+
+  constructor(apiUrl: string) {
+    this.apiUrl = apiUrl;
+  }
```

Comment:
```suggestion
  constructor(private readonly apiUrl: string) {}
```

in TS you can declare and initialise at the same time the class attributes

Reply:
That's great, much simpler. Thanks!